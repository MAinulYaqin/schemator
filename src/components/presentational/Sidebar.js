import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

import Tool from "./Tool";

const Container = styled.div`
  height: 100%;
  width: 48px;
  background: #444;
`;

const Sidebar = ({ items, onClickItem }) => (
  <Container>
    {items.map((item, index) => (
      <Tool key={index} {...item} onClick={() => onClickItem(item)} />
    ))}
  </Container>
);

Sidebar.propTypes = {
  items: PropTypes.array,
  onClickItem: PropTypes.func
};

Sidebar.defaultProps = {
  items: [],
  onClickItem: () => {}
};

export default Sidebar;
