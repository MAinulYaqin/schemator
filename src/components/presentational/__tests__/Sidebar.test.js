import React from "react";
import { shallow } from "enzyme";

import Sidebar from "../Sidebar";

const mockIcon = jest.fn();

describe("Sidebar", () => {
  const setup = propOverrides => {
    const props = {
      items: [
        {
          id: "workstage",
          icon: mockIcon,
          tooltip: "Workstage"
        },
        {
          id: "extensions",
          icon: mockIcon,
          tooltip: "Extensions"
        }
      ],
      onClickItem: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<Sidebar {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should renders with active item", () => {
    const { wrapper } = setup({ active: "extensions" });

    expect(wrapper).toMatchSnapshot();
  });

  it("should calls 'onClickItem' while item clicked", () => {
    const { wrapper, props } = setup();

    props.items.forEach((item, index) => {
      wrapper
        .find("Tool")
        .at(index)
        .simulate("click", item);

      expect(props.onClickItem).toBeCalledWith(item);
    });
  });
});
