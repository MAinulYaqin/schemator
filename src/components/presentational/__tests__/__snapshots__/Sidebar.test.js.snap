// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should renders default 1`] = `
<styled.div>
  <Tool
    icon={[Function]}
    id="workstage"
    key="0"
    onClick={[Function]}
    tooltip="Workstage"
  />
  <Tool
    icon={[Function]}
    id="extensions"
    key="1"
    onClick={[Function]}
    tooltip="Extensions"
  />
</styled.div>
`;
